clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -545,7 +545,7 @@ private void resetSequenceNumbers() {
     }
 
     synchronized void requestEpochBumpForPartition(TopicPartition tp) {
-        epochBumpRequired = true;
+        epochBumpTriggerRequired = true;
```

Comment:
As far as I can see this function is only called for idempotent (non-transactional) producers.  Can we rename the function to something like `requestIdepmpotentEpochBumpForPartition`?